compiler-core/src/build/package_compilation_tests.rs

```diff
@@ -40,8 +40,16 @@ macro_rules! assert_erlang_compile {
         let root = PathBuf::from("some/build/path/root");
         let out = PathBuf::from("_build/default/lib/the_package");
         let lib = PathBuf::from("_build/default/lib");
-        let mut compiler =
-            PackageCompiler::new(&config, &root, &out, &lib, Target::Erlang, ids, file_writer);
+        let mut compiler = PackageCompiler::new(
+            &config,
+            &root,
+            &out,
+            &lib,
+            Target::Erlang,
+            ids,
+            DependencyMode::IncludeDev,
```

Comment:
Seeing as this only does anything to Erlang generation what do you think about passing it in as part of the target? Something like this perhaps:

```rust
pub enum TargetConfiguration {
    JavaScript,
    Erlang {
        include_dev_dependencies_in_app_file: bool
    }
}
```

Otherwise I would prefer a more specific name and for it to be a property as below. I think this more general name could imply that it does more than just alter the `.app` file generate.

Reply:
@lpil What do you think of expanding the use cases for this enum? Other things I think we could skip if we knew we weren't the root project:

1. Skip downloading dev dependencies 
2. Skip compiling dev dependencies

Unless you think those _should_ happen when not the root project?

Reply:
There's no concept of a root package outside of the context of the `ProjectCompiler`, we wouldn't be able to use it in the `PackageCompiler` as it doesn't make sense there. To it all packages are equal and it knows nothing of what order things are to be compiled in.